spring-cloud-context/src/test/java/org/springframework/cloud/bootstrap/encrypt/EnvironmentDecryptApplicationInitializerTests.java

```diff
@@ -115,4 +119,88 @@ public void indexedPropertiesCopied() {
 		assertThat("decrypted property source had wrong size", decrypted.getSource().size(), is(4));
 	}
 
+	@Test
+	public void testDecryptNonStandardParent() {
+		ConfigurableApplicationContext ctx = new AnnotationConfigApplicationContext();
+		EnvironmentDecryptApplicationInitializer initializer =
+				new EnvironmentDecryptApplicationInitializer(Encryptors.noOpText());
+
+		addEnvironment(ctx, "key:{cipher}value");
+
+
+		ApplicationContext ctxParent = mock(ApplicationContext.class);
+		when(ctxParent.getEnvironment()).thenReturn(new Environment() {
```

Comment:
Not sure if that will work @spencergibb, this issue is a real corner case and can only happen if the environment is not a `ConfigurableEnvironment`. `MockEnvironment` is a `ConfigurableEnvironment` so will not simulate the bug?